image/manifest.go

```diff
@@ -0,0 +1,77 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package image
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// The seed.manifest generated by ubuntu-image contains entries in the following
+// format:
+// <snap-name> <snap-revision>.snap
+// Why they end with a .snap is anyone's guess. The goal in a future iteration of
+// this will be to move the generation of the seed.manifest to this package, out
+// of ubuntu-image.
+// TODO: Move generation of seed.manifest from ubuntu-image to here
+var revisionEntryRegex = regexp.MustCompile(`([^\s]+) ([0-9]+).snap`)
+
+// ReadSeedManifest reads a seed.manifest generated by ubuntu-image, and returns
+// an map containing the snap names and their revisions.
+func ReadSeedManifest(manifestFile string) (map[string]int, error) {
+	contents, err := ioutil.ReadFile(manifestFile)
```

Comment:
Updated the code in the loop, I think the regex provide enough protection to avoid manually parsing lines, but I added the snap.ValidateName and snap.ParseRevision